### PR TITLE
openvpn: add proto handler openvpn

### PR DESCRIPTION
**Pull request by mildred**
This is a port from https://github.com/lede-project/source/pull/1103

It is supposed to add a protocol handler for openvpn in netifd. This is needed to get the interface recognized properly, have the IP address show up in luci and supposedly other nice things.

### CODE REVIEW

**Comment by feckert** [comment]
Since we have just recently branched openwrt-19.07, i think the master is open for new ones. Let's try to push this into the master and see what the feedback is like.

File: package/network/services/openvpn/files/openvpn.init [hunk 1]
```diff
@@ -164,3 +169,56 @@ start_service() {
...
+	fi
+
+	[ ! -d "/var/etc" ] && mkdir -p "/var/etc"
+	[ -f "/var/etc/openvpn-${s}.conf" ] && rm "/var/etc/openvpn-${s}.conf"
+
+	append_bools "$s" $OPENVPN_BOOLS
+	append_params "$s" $OPENVPN_PARAMS
+
+	$log notice "Config $s generated"
+	exit 0
```
Comment (feckert) [inline package/network/services/openvpn/files/openvpn.init]: This should be $LOG

**Comment by feckert** [comment]
We should also consider https://github.com/openwrt/openwrt/pull/1596
```
--route-up /lib/netifd/openvpn-up
--route-pre-down /lib/netifd/openvpn-down
```
and use instead the hotplug.d handling from this PR

**Comment by mildred** [comment]
Just updated this PR with various fixes to make it work. In particular, the `--route-noexec` flag could cause issues, so it's left out now (and can always be manually added to the config)

A few features were added: the ability to use a different vpn name and interface name andthe ability to push IPv6 prefixes to be used on LAN

**Comment by mildred** [comment]
What is needed for this PR to be merged?

**Comment by mildred** [comment]
About hotplug, I'm not convinced this is the same thing. #1596 is about adding sonfiguration options so specific scripts could be hooked up. Here, we execute a defined script to interact with netifd. The scripts in this PR should not be put in the configuration but instead be always executed.

In any case, to use parts of #1596 in this PR, #1596 would have to be merged first.

File: package/network/services/openvpn/files/openvpn.init (hunk 1 above)
Comment (mildred) [reply package/network/services/openvpn/files/openvpn.init]: done

**Comment by adrianschmutzler** [comment]
@mildred Your patches have formal issues. Please describe the changes you do and the reasons for it. Second patch misses Signed-off-by.
Maybe have another look at https://openwrt.org/submitting-patches

**Comment by RussellSenior** [comment]
@mildred ping

**Comment by dedeckeh** [comment]
@mildred Can you address the issues outlined by @adrianschmutzler so this PR can progress ?

**Comment by wulfy23** [comment]
@mildred ping
also... is the DBG handling variation between *-up and *-down deliberate?

**Comment by adrianschmutzler** [comment]
Since there hasn't been any update by the author, I'm closing this now.

Feel free to request reopening if you want to continue working on this (note that this has to be reopened _before_ you update your corresponding branch on GitHub).

**Pull request closed without merging**